Add note to Style Guide on using Liquid comments over HTML comments (#17661)
cockroachdb/docs: StyleGuide.md

- If you intend to link to a header present on two or more tabsets on the same page, the header targets must be uniquely named. If you require identical header names, use explicit, unique HTML anchor names for each (in form `<a name="uniquename"></a>` as shown under [Links](#links).
- For the first-defined tab, specifying its `filter` value in a link is functionally the same as omitting it. For all other tabs, the explicit filter name is required. You can think of this first tab as the "default" tab in this context: if not otherwise specificed, Jekyll will always open with the first tab's contents displayed.

### Comments

There may be situations that require adding an explanation below a piece of content or to temporarily suppress content within a page. In those situations, use [Liquid comments](https://shopify.github.io/liquid/tags/template/#comment).

Page source:

```
This sentence is visible!
{% comment %}
This sentence is not visible!
{% endcomment %}

> This sentence is also {% comment %}not {% endcomment %}visible.

Reviewer: I think only the word "not" is not visible, right? Does it really hide a whole line that has a comment in the middle of it?

Author: ```
{% comment %}
This sentence is not visible!
{% endcomment %}
```
The paragraph containing `This sentence is not visible!` gets suppressed. In line 1380, only the word `not` and the space following it is suppressed. It doesn't hide the whole line. It only hides what is strictly in between the `{% comment %}` and `{% endcomment %}` tags.

> This sentence is also {% comment %}not {% endcomment %}visible.

Reviewer: ```suggestion
This sentence is visible except for a single commented word: {% comment %}not{% endcomment %}
```

```

Final page HTML:

```
<p>This sentence is visible!</p>

> <p>This setence is also visible.</p>

Reviewer: This output is no longer accurate either.

```

> Do not use HTML comments (`<!-- -->`), as the content will still be made available in the page source in production. Additionally, any HTML comment content is subject to the Liquid parser, so any Liquid inside of an HTML comment is still processed and can produce errors (e.g., an include or a link is broken).

Reviewer: ```suggestion
Do not use HTML comments (`<!-- -->`), because HTML comments are visible in the page's HTML source code in production. Additionally, any HTML comment content must be processed by the Liquid parser, so any Liquid within an HTML comment is still processed and can produce errors, such as a broken include or broken link.
```

#### TODOS

If you have future work to do in a particular file, simply add `TODO` or a `FIXME` inside of a comment. A colon after `TODO` or `FIXME` is optional.

Examples:

```
{% comment %}
TODO clean up SQL diagrams 
{% endcomment %}
```

```
{% comment %}
FIXME: Update example SQL commands
{% endcomment %}
```

> Many popular code editors feature extensions that can highlight `TODO`s across the repository. One such extension is [Todo Tree](https://marketplace.visualstudio.com/items?itemName=Gruntfuggly.todo-tree).

Reviewer: I have reservations about the allusion to profanity in the itemName. I wonder if there is another option we could point to instead.

Author: Found an alternative. It's unfortunate, because it's a good extension.
